charts/helix-controlplane/templates/deployment.yaml

```diff
@@ -79,16 +79,24 @@ spec:
               value: {{ . | quote }}
             {{- end }}
             {{- end }}
-            {{- if .Values.controlplane.inference }}
-            {{- with (default "" .Values.controlplane.inference.defaultProvider) }}
+
+            {{- with (.Values.controlplane.inference).defaultProvider }}
             - name: INFERENCE_PROVIDER
               value: {{ . | quote }}
             {{- end }}
-            {{- with (default "" .Values.controlplane.inference.defaultProvider) }}
+
+            {{- if .Values.controlplane.fineTuning }}
+            {{- with (.Values.controlplane.fineTuning).defaultProvider }}
+            - name: FINETUNING_PROVIDER
+              value: {{ . | quote }}
+            {{- end }}
+            {{- else }}
+            {{- with (.Values.controlplane.inference).defaultProvider }}
             - name: FINETUNING_PROVIDER
               value: {{ . | quote }}
             {{- end }}
             {{- end }}
+
```

Comment:
this will define FINETUNING_PROVIDER twice if both fineTuning.defaultProvider and inference.defaultProvider are set, which is not what we want. We want fineTuning.defaultProvider to overwrite inference.defaultProvider for the value of FINETUNING_PROVIDER

Reply:
sorry, missed the else and with behavior